src/logger.ts

```diff
@@ -61,7 +61,6 @@ export async function saveInterfaceStat(): Promise<void> {
   }
 
   apiPerfLogData = []
```

Comment:
I think there is still a race condition here. 
since `saveInterfaceStat()` is async, and clears the `apiPerfLogData` at the end, couldn't new data get pushed to `apiPerfLogData`  then deleted without getting logged?

Reply:
Good catch here! But instead of losing stats this can actually cause double counting, because execution is synchronous starting from when data is pushed into apiPerfLogData up until the database write. So something like this could occur:
```
fn_end[0]
  push data 0 into apiPerfLogData // [0]
  saveInterfaceStat()
    sql string from apiPerfLogData // insert 0
    START await db exec...

      fn_end[1]
        push data 1 into apiPerfLogData // [0, 1]
        saveInterfaceStat()
          sql string from apiPerfLogData // insert 0, 1
          START await db exec...

    END await db exec // wrote 0
    clear apiPerfLogData

          END await db exec // wrote 0, 1
          clear apiPerfLogData
```

Reply:
The right way to handle this is probably to clear apiPerfLogData right after creating the sql insert string but before kicking off the DB write within the same synchronous block. Then, no matter the ordering, every DB write puts something unique in when it completes and no data is lost or double counted. Will make this change and add it to the PR.